src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -151,16 +131,68 @@ func (s *fileSystemSource) AvailableIndex(
 	return s.availability(md, shardsTimeRanges)
 }
 
-func (s *fileSystemSource) ReadIndex(
-	md namespace.Metadata,
-	shardsTimeRanges result.ShardTimeRanges,
-	runOpts bootstrap.RunOptions,
-) (result.IndexBootstrapResult, error) {
-	r, err := s.read(md, shardsTimeRanges, bootstrapIndexRunType, runOpts)
-	if err != nil {
-		return nil, err
+func (s *fileSystemSource) Read(
+	namespaces bootstrap.Namespaces,
+) (bootstrap.NamespaceResults, error) {
+	results := bootstrap.NamespaceResults{
+		Results: bootstrap.NewNamespaceResultsMap(bootstrap.NamespaceResultsMapOptions{}),
 	}
-	return r.index, nil
+
+	// NB(r): Perform all data bootstrapping first then index bootstrapping
```

Comment:
Is this a temporary debug thing until it can be parallelized?

Reply:
Yeah for now since the commit log bootstrapper is much slower than all the other bootstrappers we're going to just optimize for that speeding that up first.